### PR TITLE
[Tech] Added documentation on Injectors, handled edge cases

### PR DESCRIPTION
**Pull request by smiakchilo**

### CODE REVIEW

File: docs/md/injectors.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+```
+@EToolboxList("/content/etoolbox-lists/contentList")
+private List<SimpleListItem> itemsListResource;
+
+```
+
+Note that your injected field's type can be either `List` or an array, like `SimpleListItem[]`. Concerning the type of
+items, you can specify a valid Sling model, or a *Resource*, or else *Object*. In the latter case, *Resource*-s will be
+injected in place of *Object*-s. In the snippets we use the out-of-the-box *SimpleListItem* but any valid Sling model
+will go. Generally, this injector works according to the same principles as the [ListHelper](etoolbox-lists.md) utility
```
Comment (liubou-masiuk) [inline docs/md/injectors.md:29]: Perhaps we should specify that Sling Models are supported only if they are adapted from Resource

File: docs/md/injectors.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+those with the component type "*wcm/msm/components/ghost*"), and the `NonNullPredicate` filters out invalid/null
+children.
+
+Note: the `@Children` annotation can be used with either a field, a method, or a constructor argument. When using with a
+constructor, write it like `(@Children @Named("path") List<ListItemModel> argument)` or
+else `(@Children(name = "path") @Named List<ListItemModel> argument)` and also annotate the constructor itself with `@Inject`.
+
+#### Injector for request parameters
+
+The `@RequestParam` annotation is used to inject a request parameter. The annotated member can be of type *String* or *Object*, then a value coerced to string is injected. Else, the parameter can be of type *RequestParameter*, *RequestParameter* or *RequestParameterMap* so that the corresponding objects obtained via the *SlingHttpServletRequest* could be injected.
```
Comment (liubou-masiuk) [inline docs/md/injectors.md:197]: RequestParameter is duplicated

**Review by liubou-masiuk — APPROVED** [review]

**Comment by lmasiuk** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by lmasiuk** [comment]
recheck

**Pull request merged**